### ISSUE TITLE
aarch64 support

### ISSUE THREAD

**Issue by subjectdenied** [NicolasGuilloux/blade-shadow-beta#68]
with some aarch64 android phones now being able to run mainline linux and known desktops, and with projects like librem5 and pinephone, and others in the pipeline, aarch64 support is needed to get shadow running on this types of smartphone environments

**Comment by NicolasGuilloux** [comment]
The support of ARM si not planned yet. The only application that currently support ARM is Android and maybe in the future MacOS as Mac decided to move to this arch.

I agree that it would be awesome to extend the support to ARM devices but I guess it will be not the priority, as maybe bugs are still present a block a lot of Linux users with a "regular" (ie 86x64 computers) computers.

I can see the benefit for the new laptops like the [Pine64](https://store.pine64.org/product/14%e2%80%b3-pinebook-pro-linux-laptop-ansi-us-keyboard-estimated-dispatch-in-december-2019/) more than for a phone, even if you can plug it to a screen and run a full desktop experience with a fully functionnal Linux.

As this is requested by a lot of people, I do not close this issue and assign it to the devs :)

**Comment by tcheneau** [comment]
Hello,

I read the Shadow Ghost runs a RK3399 SoC and runs on a custom Linux distro based on Yocto called "Thewav1" [1]. Am I wrong to speculate that an aarch64 based client already exists? Can we download an image of the "Thewav1" OS so that we can study it (and perhaps get the client working on our own)?

Regards,
Tony


[1]: https://www.reddit.com/r/ShadowPC/comments/9smquq/shadow_ghost_tech_specs/e8pzmgs/

**Comment by NicolasGuilloux** [comment]
Hi there!

No, you're right. The Ghost uses an aarch64 chip so the client exists. But it is optimized **a lot** for the Ghost board, so I'm not sure you can easily port it to another ARM board. At least, I'm not skilled enough to even see how to begin.

Cheers,
Nover

**Comment by tcheneau** [comment]
Thank you for your answer @NicolasGuilloux. It might not be a straightforward port indeed. Do you know if there is an public image of "Thewav1" that we can download to study it ?

**Comment by NicolasGuilloux** [comment]
No, I don't think you can get it officially without tweaking. Moreover, I don't know if it breaks the terms. Sorry :/

**Comment by NicolasGuilloux** [comment]
The CTO talked about the ARM support, and they are apparently working on a Raspberry Pi 4 application. No more information, we don't even know its actual state yet.
